db/src/urls.rs

```diff
@@ -160,14 +188,18 @@ async fn store_raw_html_in_page_snapshot(
     .fetch_one(pool)
     .await?;
 
-    let markdown_result = store_markdown(pool, result.page_snapshot_id, &cleaned_html).await?;
-    println!("Markdown result: {markdown_result:?}");
+    let (_, markdown_content) =
+        store_markdown(pool, result.page_snapshot_id, &cleaned_html).await?;
+    generate_and_store_summary(pool, result.page_snapshot_id, &markdown_content).await?;
 
     Ok(result)
 }
 
-pub async fn process_page_snapshot(pool: &PgPool, page: Page) -> color_eyre::Result<()> {
-    let outcome = store_raw_html_in_page_snapshot(pool, page).await?;
-    println!("Outcome: {outcome:?}");
+pub async fn process_page_snapshot(
+    pool: &PgPool,
+    page: Page,
+    markdown: Markdown,
+) -> color_eyre::Result<()> {
+    let _outcome = persist_article(pool, page, markdown).await?;
```

Comment:
I wonder if it would be better to just call `persist_article` inline?

Just cause that method is a bit more descriptive than `process_page_snapshot` I feel like!

Reply:
Or maybe what would be better is to take these bits of `persist_article` and move them into this method!

```
 store_markdown(pool, result.page_snapshot_id, &cleaned_html).await?;
    generate_and_store_summary(pool, result.page_snapshot_id, &markdown_content).await?;
```

That way the `persist_artcile` method doesn't _also_ have to create the summary and persist it!

Overall though I might still remove `process_page_snapshot` just cause the things it calls are nicely named, and I don't think this function is really helping us much 🤷